unlock-app/src/components/interface/checkout/InsufficientFundsWarning.tsx

```diff
@@ -1,22 +1,208 @@
+  const [showFundingContent, setShowFundingContent] = useState(false)
+  const web3Service = useWeb3Service()
+
+  const { recipients, paywallConfig, keyManagers, renew } = context
+
+  const { data: baseRoute } = useQuery({
+    queryKey: ['prices', userAddress, lock, recipients, purchaseData],
+    queryFn: async () => {
+      if (!purchaseData || !userAddress || !lock || !recipients || renew) {
+        return []
+      }
+
+      const prices = await purchasePriceFor(web3Service, {
+        lockAddress: lock.address,
+        network: lock.network,
+        recipients,
+        data: purchaseData || recipients.map(() => ''),
+        paywallConfig,
+        currencyContractAddress: lock.currencyContractAddress,
+        symbol: lock.currencySymbol,
+      })
+
+      const price = prices.reduce((acc, item) => acc + item.amount, 0)
+
+      if (isNaN(price) || price === 0) {
+        return []
+      }
+
+      const sharedParams = await prepareSharedParams({
+        lock,
+        prices: prices!,
+        recipients,
+        keyManagers: keyManagers || recipients,
+        referrers: recipients.map(() =>
+          getReferrer(userAddress, paywallConfig, lock.address)
+        ),
+        purchaseData,
+      })
+
+      const route = await relayGetCrossChainRoute({
+        sender: userAddress,
+        lock,
+        prices: prices!,
+        srcToken: ADDRESS_ZERO,
+        recipients,
+        keyManagers: keyManagers || recipients,
+        referrers: recipients.map(() =>
+          getReferrer(userAddress, paywallConfig, lock.address)
+        ),
+        purchaseData,
+        // always use the base chain
+        srcChainId: 8453,
+        sharedParams: sharedParams!,
+      })
+
+      return route
+    },
+    staleTime: 1000 * 60 * 5,
+  })
+
+  const { fundWallet } = useFundWallet({
+    onUserExited: async ({ balance }) => {
+      // Format the balance from bigint to decimal string
+      const formattedBalance = balance
+        ? ethers.formatEther(balance.toString())
+        : '0'
+
+      // Get route amount based on the payment currency
+      const requiredAmountInETH =
+        baseRoute &&
+        'tokenPayment' in baseRoute &&
+        baseRoute.tokenPayment?.amount
+          ? parseFloat(
+              ethers.formatEther(baseRoute.tokenPayment.amount)
+            ).toFixed(4)
+          : requiredAmount
+
+      if (Number(formattedBalance) < Number(requiredAmountInETH)) {
+        handleSetShowFundingContent(false)
+        onShowFundingContent?.(false)
+        return
+      }
+
+      checkoutService.send({
+        type: 'SELECT_PAYMENT_METHOD',
+        payment: {
+          method: 'crosschain_purchase',
+          route: baseRoute,
+        },
+      })
+
+      handleSetShowFundingContent(false)
+      onShowFundingContent?.(false)
+    },
+  })
+
+  const handleFundWallet = async () => {
+    // Use route amount for any currency, with a small buffer for gas
+    const fundingAmount =
```

Comment:
Same comment as above. If `baseRoute.tokenPayment?.amount` is falsy, let's fail rather than use `requiredAmount` (I don' understand what it is)

Also, let's not use `toFixed` here because it rounds the number... and if it rounds below we have an issue!

Also, let's create a variable `gasCost = 0.0001` with a comment indicating that it would be nice to have that from relay so we don't have to hardcode it.